### PR TITLE
Add tests for navigate on form submit

### LINKED ISSUE #21241: Blur event inside form can cause "no event handler associated with this event" exception
Scenario:

```razor
<EditForm OnValidSubmit="NavigateToAnotherPage">
    <input @blur="@(() => {})" />
</EditForm>
```

Pressing "enter" inside the `input` causes:

```
System.ArgumentException: There is no event handler associated with this event. EventId: '45'.
Parameter name: eventHandlerId
  at Microsoft.AspNetCore.Components.RenderTree.Renderer.DispatchEventAsync (:44300/System.UInt64 eventHandlerId, Microsoft.AspNetCore.Components.RenderTree.EventFieldInfo fieldInfo, System.EventArgs eventArgs) <0x3124c68 + 0x00054> in <filename unknown>:0 
  at Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer.DispatchEventAsync (:44300/System.UInt64 eventHandlerId, Microsoft.AspNetCore.Components.RenderTree.EventFieldInfo eventFieldInfo, System.EventArgs eventArgs) <0x31248a0 + 0x00078> in <filename unknown>:0 
  at :44300/Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer.ProcessNextDeferredEventAsync () <0x3477c48 + 0x000a6> in <filename unknown>:0 
    at Object.endInvokeDotNetFromJS (blazor.webassembly.js:1)
    at Object.invokeJSFromDotNet (blazor.webassembly.js:1)
```

Looks like the "navigate away" gets processed before the "blur", so the component has been disposed by the time we process its "blur" event handler.

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/dotnet/aspnetcore/issues/21241.

I wasn't able to repro the issue on master so it looks like something we've done between now and then has resolved the issue.

Adding a test for this bug so we have coverage for it.

### CODE REVIEW

**Comment by SteveSandersonMS** [comment]
I was a bit thrown off by this because I didn't think we'd changed anything that would be likely to affect this. Upon verifying, this does still repro for me on `master`. Is it possible that it doesn't repro within the E2E project because it doesn't have a top-level `<Router>`, and hence it's trying to do server-side navigation?

For clarification here's what reproduces the issue for me: https://github.com/dotnet/aspnetcore/compare/stevesa/blur-bug-repro. Apologies for the ambiguity/typos in the original bug report.

**Comment by captainsafia** [comment]
> I was a bit thrown off by this because I didn't think we'd changed anything that would be likely to affect this. Upon verifying, this does still repro for me on master. Is it possible that it doesn't repro within the E2E project because it doesn't have a top-level <Router>, and hence it's trying to do server-side navigation?

Hm. I'll check on this. I validated inside the test harness and in a new standalone WASM project using 5.0-dev bits. I'll give your repro ago in case I missed something.

**Comment by captainsafia** [comment]
@SteveSandersonMS Woop. You're right. I'm able to repro this on a standalone app with the `5.0.100-rc.1.20405.4` SDK so it's definitely still an issue. I'll investigate and push a fix to this PR.

**Review by SteveSandersonMS — APPROVED** [review]
Great!

File: src/Components/Web.JS/src/Rendering/EventDelegator.ts [hunk 1]
```diff
@@ -170,6 +174,10 @@ class EventInfoStore {
     this.addGlobalListener(info.eventName);
   }
 
+  public get(eventHandlerId: number) {
```
Comment (campersau) [inline src/Components/Web.JS/src/Rendering/EventDelegator.ts:177]:
```suggestion
  public get(eventHandlerId: number): EventHandlerInfo {
```
like in `public remove(eventHandlerId: number): EventHandlerInfo {`

Comment (SteveSandersonMS) [reply src/Components/Web.JS/src/Rendering/EventDelegator.ts:177]: Reasonable suggestion (thanks @campersau!), but this is completely internal code so this change wouldn't really affect anything. We can change it further at any time.

Comment (captainsafia) [reply src/Components/Web.JS/src/Rendering/EventDelegator.ts:177]: Yeah, I think TypeScript's contextual typing will take care of implicitly setting the correct type here.

**Pull request merged**